### PR TITLE
Web servers intro, got most help from the mdn docs

### PR DESCRIPTION
**Pull request by potato-Js**
#### What roadmap does this PR target?

- [ ] Code Change
- [ ] Frontend Roadmap
- [x] Backend Roadmap
- [ ] DevOps Roadmap
- [ ] All Roadmaps
- [ ] Guides

#### Please acknowledge the items listed below

- [x] I have discussed this contribution and got a go-ahead in an issue before opening this pull request.
- [x] This is not a duplicate issue. I have searched and there is no existing issue for this.
- [x] I understand that these roadmaps are highly opinionated. The purpose is to not to include everything out there in these roadmaps but to have everything that is most relevant today comparing to the other options listed.
- [x] I have read the [contribution docs](../contributing) before opening this PR.

#### Enter the details about the contribution

wrote a simple intro about webservers with help from the mdn docs and included a link there also with a youtube video which is pretty good

### CODE REVIEW

File: content/roadmaps/101-backend/content/122-web-servers/readme.md [hunk 1]
```diff
@@ -1 +1,16 @@
...
+A web server has a number of software components that regulate how hosted files are accessed by online users. This is at the very least an HTTP server. Software that knows and understands HTTP and URLs (web addresses) is known as an HTTP server (the protocol your browser uses to view webpages). The content of these hosted websites is sent to the end user's device through an HTTP server, which may be accessed via the domain names of the websites it holds.
+ 
+basically, an HTTP request is made by a browser anytime it wants a file that is stored on a web server. The relevant (hardware) web server receives the request, which is then accepted by the appropriate (software) HTTP server, which then locates the requested content and returns it to the browser over HTTP. (If the server cannot locate the requested page, it responds with a 404 error.)
+
+<ResourceGroupTitle>Free Content</ResourceGroupTitle>
+<BadgeLink colorScheme='blue' badgeText='Official Website' href='https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_is_a_web_server' >MDN DOCS </BadgeLink>
+<BadgeLink badgeText='Watch' href='https://youtu.be/9J1nJOivdyw'>Web Server Concepts and Examples</BadgeLink>
```
Comment (kamranahmedse) [inline content/roadmaps/101-backend/content/122-web-servers/readme.md:16]:
```suggestion
# Web Servers

Web servers can be either hardware or software, or perhaps a combination of the two. 

### Hardware Side:

A hardware web server is a computer that houses web server software and the files that make up a website (for example, HTML documents, images, CSS stylesheets, and JavaScript files). A web server establishes a connection to the Internet and facilitates the physical data exchange with other web-connected devices.

### Software side:

A software web server has a number of software components that regulate how hosted files are accessed by online users. This is at the very least an HTTP server. Software that knows and understands HTTP and URLs (web addresses) is known as an HTTP server (the protocol your browser uses to view webpages). The content of these hosted websites is sent to the end user's device through an HTTP server, which may be accessed via the domain names of the websites it holds.
 
Basically, an HTTP request is made by a browser anytime it wants a file that is stored on a web server. The relevant (hardware) web server receives the request, which is then accepted by the appropriate (software) HTTP server, which then locates the requested content and returns it to the browser over HTTP. (If the server cannot locate the requested page, it responds with a 404 error.)

<ResourceGroupTitle>Free Content</ResourceGroupTitle>
<BadgeLink colorScheme='yellow' badgeText='Read' href='https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_is_a_web_server' >What is a Web Server </BadgeLink>
<BadgeLink badgeText='Watch' href='https://youtu.be/9J1nJOivdyw'>Web Server Concepts and Examples</BadgeLink>
```
